internal/services/hdinsight/hdinsight_spark_cluster_resource_test.go

```diff
@@ -1788,3 +1808,265 @@ resource "azurerm_hdinsight_spark_cluster" "test" {
+  }
+
+  security {
+    ntlm_v1_enabled         = true
+    sync_kerberos_passwords = true
+    sync_ntlm_passwords     = true
+    sync_on_prem_passwords  = true
+    tls_v1_enabled          = true
+  }
+
+  tags = {
+    Environment = "test"
+  }
+
+  depends_on = [
+    azuread_group_member.test,
+    azurerm_role_assignment.test,
+    azurerm_subnet_network_security_group_association.test,
+  ]
+}
+
+resource "azurerm_virtual_network_dns_servers" "test" {
+  virtual_network_id = azurerm_virtual_network.test.id
+  dns_servers        = azurerm_active_directory_domain_service.test.initial_replica_set.0.domain_controller_ip_addresses
+}
+
+resource "azurerm_hdinsight_spark_cluster" "test" {
+  name                = "acctesthdispark-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  cluster_version     = "4.0"
+  tier                = "Premium"
+
+  component_version {
+    spark = "2.4"
+  }
+
+  gateway {
+    username = "sshuser"
+    password = "TerrAform123!"
+  }
+
+  storage_account {
+    storage_container_id = azurerm_storage_container.test.id
+    storage_account_key  = azurerm_storage_account.test.primary_access_key
+    is_default           = true
+  }
+
+  roles {
+    head_node {
+      vm_size            = "Standard_E8_V3"
+      username           = "sshuser"
+      password           = "TerrAform123!"
+      subnet_id          = azurerm_subnet.test.id
+      virtual_network_id = azurerm_virtual_network.test.id
+    }
+
+    worker_node {
+      vm_size               = "Standard_E8_V3"
+      username              = "sshuser"
+      password              = "TerrAform123!"
+      target_instance_count = 3
+      subnet_id             = azurerm_subnet.test.id
+      virtual_network_id    = azurerm_virtual_network.test.id
+    }
+
+    zookeeper_node {
+      vm_size            = "Standard_A2_V2"
+      username           = "sshuser"
+      password           = "TerrAform123!"
+      subnet_id          = azurerm_subnet.test.id
+      virtual_network_id = azurerm_virtual_network.test.id
+    }
+  }
+
+  security_profile {
+    aadds_resource_id       = "${replace(azurerm_active_directory_domain_service.test.id, "/initialReplicaSetId/${azurerm_active_directory_domain_service.test.deployment_id}", "")}"
+    domain_name             = azurerm_active_directory_domain_service.test.domain_name
+    domain_username         = azuread_user.test.user_principal_name
+    domain_user_password    = azuread_user.test.password
+    ldaps_urls              = ["ldaps://${azurerm_active_directory_domain_service.test.domain_name}:636"]
+    msi_resource_id         = azurerm_user_assigned_identity.test.id
+    cluster_users_group_dns = [azuread_group.test.display_name]
+  }
+
+  depends_on = [
+    azurerm_virtual_network_dns_servers.test,
+  ]
+
+  lifecycle {
+    ignore_changes = [
+      security_profile.0.domain_user_password,
+      gateway.0.password
+    ]
```

Comment:
As above

Reply:
Service API doesn't return the values for these properties. So we have to ignore them. Otherwise, we would encounter diff while running these TCs.

Reply:
OK. Thanks. Updated.